tools/env-loader/pkg/writers/gha-env.go

```diff
@@ -0,0 +1,103 @@
+/*
+ *  Copyright 2024 Gravitational, Inc
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package writers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/gravitational/shared-workflows/tools/env-loader/pkg/values"
+	"github.com/gravitational/trace"
+)
+
+const delimiterPrefix = "EOF"
+
+// Outputs values in a format that can be parsed by GHA's `GITHUB_ENV` file.
+// This is _almost_ the same as dotenv files, but also handles multiline
+// environment values. For details, see
+// https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-environment-variable
+type GHAEnvWriter struct{}
+
+// Create a new GHA env writer
+func NewGHAEnvWriter() *GHAEnvWriter {
+	return &GHAEnvWriter{}
+}
+
+// Generates a delimiter that is guaranteed to not contain the provided string.
+// This is required for writing multiline values to `GITHUB_ENV`, per
+// https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#multiline-strings
+func generateMultilineDelimiter(value string) string {
```

Comment:
Since you're using UUID in your delimiter string, the chance of collision is negligible and this function is needlessly complicated and can be reduced to literally just

```go
func getDelimiter() string {
    return delimiterPrefix + "_" + uuid.NewString()
}
```

Maybe even no need for the function at all and just do it inline.